annotations/builder/src/main/java/io/sundr/builder/internal/utils/BuilderUtils.java

```diff
@@ -463,17 +465,48 @@ public static Property buildableField(Property property) {
                     .addToAttributes(INIT, " new " + listRef + "()")
                     .addToAttributes(ALSO_IMPORT, alsoImport(property, listRef, builderType))
                     .build();
-        } else if (TypeUtils.isSet(classRef)) {
+        }
+
+        if (TypeUtils.isSet(classRef)) {
             ClassRef setRef = Collections.LINKED_HASH_SET.toReference(builderType);
             return new PropertyBuilder(property).withTypeRef(Collections.SET.toReference(builderType))
                     .addToAttributes(INIT, " new " + setRef+ "()")
                     .addToAttributes(ALSO_IMPORT,  alsoImport(property, setRef, builderType))
                     .build();
-        } else {
-            return new PropertyBuilder(property).withTypeRef(builderType)
+        }
+
+        if (TypeUtils.isOptionalLong(classRef)) {
+            ClassRef optionalRef = Optionals.OPTIONAL_LONG.toReference(builderType);
+            return new PropertyBuilder(property).withTypeRef(optionalRef)
+                .addToAttributes(INIT, " " + optionalRef + ".empty()")
```

Comment:
There is no value in using the actual ref (as it causes issues in many cases). Just use OptionalLong.empty(). Easier and safer.